src/modules/hip/kernel/image_sum.hpp

```diff
@@ -0,0 +1,480 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - Reduction Stage 2 --------------------
+
+__global__ void image_sum_grid_result_tensor(float *srcPtr,
+                                             uint xBufferLength,
+                                             float *dstPtr)
+{
+    int id_x = hipThreadIdx_x * 8;
+    int id_z = hipBlockIdx_z;
+
+    __shared__ float partialSumLDS[256];                            // 1024 floats of src reduced to 256 in a 256 x 1 thread block
```

Comment:
This should probably be 2048 floats, we can also increase lds or threads to accomodate a larger image. Lets discuss.

Reply:
Done, modified it to 1024 and we are able to support upto 4k image sizes